Kconfig.dependencies

```
@@ -114,26 +115,42 @@ config SIDEWALK_LOG
 	help
 	  Sidewalk log module
 
-config SIDEWALK_GENERATE_VERSION_MINIMAL
-	bool
-	default SIDEWALK
-	help
-	  Generate minimal information of application version (sidewalk, zephyr, nrf).
-	  In order to generate full version report from all modules set this config to `N`.
+choice SIDEWALK_MFG_STORAGE
+	prompt "Select implementation of mfg_storage"
+	default SIDEWALK_MFG_STORAGE_V1
 
-config SIDEWALK_MFG_STORAGE
-	bool
-	default SIDEWALK
+config SIDEWALK_MFG_STORAGE_V1
+	bool "Use mfg_storage v1"
 	imply FLASH
 	help
-	  Sidewalk manufacturing storage module
+	  Sidewalk manufacturing storage module v1
+
+config SIDEWALK_MFG_STORAGE_V2
+	bool "Use mfg_storage v2"
+	imply WARN_EXPERIMENTAL
+	imply FPROTECT
+	imply SIDEWALK_TLV
+	imply SIDEWALK_TLV_FLASH
+	imply SIDEWALK_TLV_RAM
+	help
+	  Sidewalk manufacturing storage module v2
+endchoice
 
-config SIDEWALK_MFG_STORAGE_WRITE
+config SIDEWALK_MFG_STORAGE_DIAGNOSTIC
 	bool
 	default SIDEWALK_ON_DEV_CERT
-	depends on SIDEWALK_MFG_STORAGE
 	help
-	  Enable write functionality to mfg storage
+	  Enable mfg storage diagnostic functionalities 
```

Keep it consistent in another configs description
```suggestion
	  Enable manufacturing storage diagnostic functionalities 
```
The same in next configs.